src/plugins/validate-semantic/validators/2and3/schemas.js

```diff
@@ -23,41 +23,46 @@ export const validate2And3TypeArrayRequiresItems = () => (system) => {
     })
 }
 
+
+
 export const validate2And3TypesInDefaultValuesMatchesWithEnum = () => (system) => {
   return system.validateSelectors
     .allSchemas()
     .then(nodes => {
       return nodes.reduce((acc, node) => {
         const schemaObj = node.node
         const { type } = schemaObj || {}
+        const isNullable = schemaObj.nullable ? schemaObj.nullable : false
         const enumeration = schemaObj.enum
         if (enumeration !== null && typeof enumeration !== "undefined") {
           var enumIndex = 0
           enumeration.forEach((element, index) => {
             var isValidFormat = true
-            if (type === "array" && (!Array.isArray(element) || element === null)) {
-              isValidFormat = false
-              enumIndex = index
-            } else if ((type === "number" || type === "string" || type === "boolean") && !(typeof element === type)) {
-              isValidFormat = false
-              enumIndex = index
-            } else if (type === "integer" && !Number.isInteger(element)) {
-              isValidFormat = false
-              enumIndex = index
-            } else if (type === "object" && ((element === null) || !(typeof element === type) || Array.isArray(element))) {
-              isValidFormat = false
-              enumIndex = index
-            }
-            if (!isValidFormat) {
-              acc.push({
-                message: "enum value should conform to its schema's `type`",
-                path: [...node.path, "enum", enumIndex],
-                level: "warning",
-              })
+            if ((element === null & !isNullable) || element !== null) {
```

Comment:
This conditional is a bit confusing to follow, and adds to the already-not-great indentation level in this validator.

You can achieve equivalent functionality by returning from the `forEach` iterator function:

```suggestion
            if (element === null && isNullable) {
              return
            }
```

This pattern is common in JavaScript, and is roughly equivalent to a `continue` statement in a conventional loop:

```js
const arr = [1, 2, 3, 4, 5]

// both of the loops below log the same four lines: "1", "3", "4", "5"

for(var num of arr) {
  if(num === 2) {
    continue
  }
  console.log(num)
}

arr.forEach(num => {
  if(num === 2) {
    return
  }
  console.log(num)
}
```

(if you commit this, don't forget to pull in the suggestion to delete the closing bracket below!)

Reply:
Sweeet! Thanks!